src/activity_handlers.cpp

```diff
@@ -2538,6 +2540,13 @@ void activity_handlers::mend_item_finish( player_activity *act, player *p )
     p->invalidate_crafting_inventory();
 
     target->faults.erase( *f );
+    std::string thefault = act->name.c_str();
```

Comment:
```suggestion
    std::string thefault = act->name;
```
But why do you even need this variable? Just compare `act->name` directly.